testutils/infra/container_manager/factory.py

```diff
@@ -0,0 +1,49 @@
+
+from .docker_compose_manager import DockerComposeStandardSetup
+from .docker_compose_manager import DockerComposeDockerClientSetup
+from .docker_compose_manager import DockerComposeRofsClientSetup
+from .docker_compose_manager import DockerComposeLegacyClientSetup
+from .docker_compose_manager import DockerComposeSignedArtifactClientSetup
+from .docker_compose_manager import DockerComposeShortLivedTokenSetup
+from .docker_compose_manager import DockerComposeFailoverServerSetup
+from .docker_compose_manager import DockerComposeEnterpriseSetup
+from .docker_compose_manager import DockerComposeEnterpriseSmtpSetup
+
+class ContainerManagerFactory:
+    def getStandardSetup(self, name, num_clients=1): pass
+    def getDockerClientSetup(self, name): pass
```

Comment:
Question: 
It sort of stands out to me that the generic container manager has docker in its API.
Is this done on purpose?

Reply:
Good eyes! :eyes: I see your point and my first reaction was to agree with you in that it is wrong.

However, looking a bit closer, the point is that the container image used in this setup is called `mender-client-docker` (as per being different than `mender-client-qemu` or `mender-client-qemu-rofs`). And I think that even in the upcoming migration to a different container orchestration, the container image itself is unlikely to change. So unless we rename the image, then I think that "Docker" belongs here.

@kacf @tranchitella Do you guys think that we should rename the image itself?

Reply:
Indeed, the "docker" here is a reference to the specific client image which is a pure docker image, not the bastardized VM-in-Docker solution which is the other image. We could consider renaming it maybe, but every name I can think of just makes it less clear.